addons/web_editor/controllers/main.py

```diff
@@ -297,10 +297,17 @@ def _attachment_create(self, name='', data=False, url=False, res_id=False, res_m
         if data:
             attachment_data['datas'] = data
         elif url:
-            attachment_data.update({
-                'type': 'url',
-                'url': url,
-            })
+            # Fetch the HEAD of the URL and retrieve the mimetype
+            response = requests.head(url)
+            if response.status_code == 200:
+                content_type = response.headers.get('content-type')
+                attachment_data.update({
+                    'type': 'url',
+                    'url': url,
+                    'mimetype': content_type  # Add mimetype obtained from HEAD request
+                })
+            else:
+                raise UserError(_("Error fetching data from URL: %s") % url)
```

Comment:
@SimonGenin What should be the expected behavior when the request cannot proceed ? In JS we decided to let it go and just not specify the mimetype. Here in Python @shsa-odoo decided to stop the uploading. I feel like it might make more sense to let it go and let the server decide the mimetype as a fallback.

On one hand it might just be some issue in the network at that time and I don't think it should prevent the uploading, especially if the file extension would be enough to guess the mimetype correctly. On the other hand, letting it continue and guess the mimetype means that a temporary network error would result in different mimetypes for the same upload if the file has a mimetype that couldn't be guessed properly from the extension, which would be kind of weird and unexpected. One time you use an url and it's correctly recognized as an image and the other time you use the very same url but somehow it's not recognized as an image this time just because there was some random network error at the wrong time.

I think I'm leaning towards blocking the upload as it would make more sense to the user. They are uploading from a URL, there is a network error, ok they will probably retry. Whereas having a message telling them that the URL is not an image... they might not retry as they might think it's never going to work, while in fact it's just bad luck that the network had a random failure and we ended up guessing the mimetype wrong. 

What is your opinion ?

Reply:
@dmo-odoo well i did it for the scenario where the MIME type is valid, but the actual data content is not found or unavailable. This is an important consideration when dealing with data fetched from URLs.

Fetching the MIME type through a HEAD request is a useful step to determine the type of data expected from the URL, but it doesn't guarantee that the data is present and accessible. There are cases where the server might respond with a valid status code (e.g., 200 OK) but provide an empty response body or indicate that the resource is not found. In such situations, relying solely on the presence of a valid MIME type might not be sufficient.

Reply:
@dmo-odoo I agree with the solution where you don't upload if the head request failed. It just makes more sense for several reasons, including the ones you gave.